### PR TITLE
docs(date-picker): add labels to website demos

### LINKED ISSUE #3465: Improve date picker documentation for labels
## Describe the bug

Our date picker docs rarely use labels with their demos, and should always have a label for accessibility reasons.

## How to reproduce

See website docs https://clarity.design/documentation/datepicker

## Expected behavior

Our documentation demos and examples should be improved.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

### PR DESCRIPTION
**Pull request by hippee-lee**
This adds labels to the demos to demonstrate proper use of clr-date-containers and labels for accessible forms.
closes #3465

Signed-off-by: Matt Hippely <mhippely@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?
Current datepicker demos and example code show using `clrDate` without a label. 

Issue Number: N/A

## What is the new behavior?
This change adds clr-date-container and labels to the datepicker demos. 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
 <form clrForm clrLayout="vertical">
   <clr-date-container>
+    <label for="minDemo">Min Demo</label>
     <input type="date" clrDate name="demo" [(ngModel)]="demo" min="2019-11-11">
   </clr-date-container>
 </form>
 `;
 const MAX_EXAMPLE = `
-  <clr-date-container>
+  <clr-date-container clrLayout="vertical">
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts]: `clrLayout` doesn't apply here, it only goes on the `form[clrForm]` element, please remove these.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.html [hunk 2]
```diff
@@ -8,11 +8,12 @@
 </clr-alert>
 <form clrForm class="datepicker-demo" clrLayout="vertical">
     <clr-date-container>
+        <label for="dateObject">Date Object</label>
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.html]: No `for` should be provided in the demos unless you explicitly define the `id` and want to (like for testing). Should remove them from our docs and samples.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html [hunk 3]
```diff
@@ -37,11 +38,24 @@ <h4>Min/Max attributes</h4>
...
-        <input type="date" clrDate name="minDemo" [(ngModel)]="demo" min="2019-11-11">
+        <label for="minDemo">Min Demo</label>
+        <input type="date" clrDate name="minDemo" [(ngModel)]="demo" min="2019-11-17">
     </clr-date-container>
 </form>
 <clr-code-snippet [clrCode]="minExample"></clr-code-snippet>
 <h5>Only Max</h5>
+<form clrForm clrLayout="vertical">
+    <clr-date-container>
+        <label for="maxDemo">Max Demo</label>
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html]: Let's also put the min/max dates in the label so people know what it is more easily.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts [hunk 4]
```diff
@@ -1,33 +1,37 @@
...
+        <input id="basicDemo" type="date" clrDate name="demo" [(ngModel)]="demo">
+    </clr-date-container>
 </form>
 `;
 
 const MIN_EXAMPLE = `
 <form clrForm clrLayout="vertical">
   <clr-date-container>
+    <label for="minDemo">Min Demo</label>
     <input type="date" clrDate name="demo" [(ngModel)]="demo" min="2019-11-11">
```
Comment (bdryanovski) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts]: Missing `id="minDemo"`

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts [hunk 5]
```diff
@@ -1,33 +1,37 @@
...
+    <label for="maxDemo">Max Demo</label>
     <input type="date" clrDate name="demo" [(ngModel)]="demo" max="2019-11-11">
   </clr-date-container>
 </form>
 `;
 const MIN_MAX_EXAMPLE = `
-  <clr-date-container>
+  <clr-date-container clrLayout="horizontal">
+    <label for="minMaxDemo">Max Demo</label>
     <input type="date" clrDate name="demo" [(ngModel)]="demo" min="2019-11-11" max="2019-12-12">
```
Comment (bdryanovski) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts]: Missing `id="minMaxDemo"`

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html [hunk 6]
```diff
@@ -37,11 +38,24 @@ <h4>Min/Max attributes</h4>
 <h5>Only Min</h5>
 <form clrForm clrLayout="vertical">
     <clr-date-container>
-        <input type="date" clrDate name="minDemo" [(ngModel)]="demo" min="2019-11-11">
+        <label for="minDemo">Min Demo</label>
+        <input type="date" clrDate name="minDemo" [(ngModel)]="demo" min="2019-11-17">
```
Comment (bdryanovski) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html]: Missing `id="minDemo"`

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts (hunk 4 above)
Comment (Jinnie) [reply src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts]: This probably relates to Jeremy's comments that all these "for"s should better go away.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts (hunk 1 above)
Comment (hippee-lee) [reply src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts]: I will remove.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.html (hunk 2 above)
Comment (hippee-lee) [reply src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.html]: removed.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html (hunk 3 above)
Comment (hippee-lee) [reply src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html]: ok

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html [hunk 7]
```diff
@@ -5,9 +5,10 @@ <h4><code class="clr-code">clrDate</code> Directive</h4>
     <code class="clr-code">input</code> field. Then, place the input inside the
     <code class="clr-code">clr-date-container</code> container element.
 </p>
-<form class="datepicker-demo" clrForm clrLayout="vertical">
+<form class="datepicker-demo" clrForm clrLayout="horizontal">
     <clr-date-container>
-        <input type="date" clrDate name="demo" [(ngModel)]="demo">
+        <label for="basicDemo">Basic Demo</label>
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html]: Need to remove `for` and `id` from all demos, sorry if that wasn't clear.

Comment (hippee-lee) [reply src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html]: I see, I will remove these as well.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts [hunk 8]
```diff
@@ -1,34 +1,40 @@
...
+    <label>Max date: 2019-11-19</label>
+    <input type="date" clrDate name="demo" [(ngModel)]="demo" max="2019-11-19">
   </clr-date-container>
 </form>
 `;
 const MIN_MAX_EXAMPLE = `
+<form clrForm>
   <clr-date-container>
-    <input type="date" clrDate name="demo" [(ngModel)]="demo" min="2019-11-11" max="2019-12-12">
+    <label>Min date: 2019-11-17 AND Max date: 2019-11-19</label>
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.ts]: Last nit: when you squash copy these labels into the actual demos above so they are the same.

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.html (hunk 2 above)
Comment (gnomeontherun) [reply src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.html]: Still here?

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.ts [hunk 9]
```diff
@@ -8,7 +8,8 @@ import { Component } from '@angular/core';
 const HTML_EXAMPLE = `
 <form clrForm clrLayout="vertical">
   <clr-date-container>
-    <input type="date" [(clrDate)]="date">
+    <label for="dateObject">Date Object</label>
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-date-io.demo.ts]: `for` and `id` remove

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-enUS.demo.ts [hunk 10]
```diff
@@ -10,7 +10,10 @@ import { Component } from '@angular/core';
   template: `
         <h5 style="margin-top: 24px">Locale Identifier: en-US</h5>
         <form clrForm clrLayout="vertical">
-            <input type="date" clrDate>
+            <clr-date-container>
+                <label for="usLocale">US locale</label>
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-enUS.demo.ts]: `for` and `id`

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-fr.demo.ts [hunk 11]
```diff
@@ -10,7 +10,10 @@ import { Component, LOCALE_ID } from '@angular/core';
   template: `
         <h5 style="margin-top: 24px">Locale Idenitifer: fr</h5>
         <form clrForm clrLayout="vertical">
-            <input type="date" clrDate>
+            <clr-date-container>
+                <label for="frLocale">FR Locale</label>
```
Comment (gnomeontherun) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-fr.demo.ts]: `for` and `id`

**Pull request merged**